src/config.rs

```diff
@@ -83,17 +71,8 @@ impl Config {
         node_url: Option<String>,
         db_url: Option<String>,
     ) -> Result<(Self, PathBuf)> {
-        let toml = fs::read_to_string(path)?;
-        let mut config: Config = toml::from_str(&toml)?;
-        // Manual Overrides from env vars.
-        if let Some(ethrpc) = &node_url {
-            tracing::info!("using env NODE_URL");
-            config.set_ethrpc(Url::parse(&ethrpc)?)
-        }
-        if let Some(connection) = &db_url {
-            tracing::info!("using env DB_URL");
-            config.set_database(connection.to_string())
-        }
+        let toml = manual_override(fs::read_to_string(path)?, node_url, db_url)?;
+        let config: Config = toml::from_str(&toml.to_string())?;
```

Comment:
Oh, I didn't know that this would happen. I tried finding the code to circumvent the serialization + deserialization, but it's not too obvious from the toml crate docs.

Reply:
It isn't, I was not able to find this yet so at least we have something that works smoothly and looks pretty for now.

Reply:
Created an issue here: #12 Will cross reference these comments.